### PR TITLE
PLATOPS-2088

### PR DESCRIPTION
**Pull request by jennygrahamjones**

### CODE REVIEW

**Comment by konradwudkowski** [comment]
Could you please format with scalafmt?

File: app/uk/gov/hmrc/slacknotifications/model/NotificationRequest.scala [hunk 1]
```diff
@@ -25,7 +25,14 @@ final case class MessageDetails(
                                  username: String,
                                  iconEmoji: Option[String] = None,
                                  attachments: Seq[Attachment] = Nil
-                               )
+                               ) {
+
+  def getFields: Array[String] = getClass.getDeclaredFields.map(field => {
```
Comment (konradwudkowski) [inline app/uk/gov/hmrc/slacknotifications/model/NotificationRequest.scala]: Is reflection necessary here? There are only 4 fields that seem to be known at compile time

File: app/uk/gov/hmrc/slacknotifications/model/SlackMessage.scala [hunk 2]
```diff
@@ -37,15 +37,41 @@ case class Attachment(
...
+      this.title,
+      this.title,
+      this.text,
+      this.image_url,
+      this.thumb_url,
+      this.footer,
+      this.footer_icon
+    )
+
+    val additionalFields: Array[String] = if(this.fields.isDefined) this.fields.map(fieldSeq => fieldSeq.flatMap(_.fields).toArray).get else Array[String]()
```
Comment (konradwudkowski) [inline app/uk/gov/hmrc/slacknotifications/model/SlackMessage.scala]: the `if(option.isDefined) option.get` is not very idiomatic in scala. Could you maybe `.getOrElse`, `.fold` or pattern match instead?

File: app/uk/gov/hmrc/slacknotifications/services/AuthService.scala [hunk 3]
```diff
@@ -58,6 +59,37 @@ class AuthService @Inject()(configuration: Configuration) {
     } else {
       true
     }
+
+  def isAuthorizedUrl(url: String): Boolean = authConfiguration.authorizedUrls.contains(url)
+
+  def filterFieldsForURLs(fields: Array[String]): Array[String] = {
+    val pattern =
+      ("(?:[a-zA-Z]+:)?(?:\\/\\/)?(?:[\\w-]+:[\\w-]+@)?(?:[\\w-]+\\.)+[a-zA-Z]+" +
```
Comment (konradwudkowski) [inline app/uk/gov/hmrc/slacknotifications/services/AuthService.scala]: it's virtually impossible to tell what this regex checks, maybe the variable name can provide hints?

File: app/uk/gov/hmrc/slacknotifications/model/SlackMessage.scala (hunk 2 above)
Comment (jennygrahamjones) [reply app/uk/gov/hmrc/slacknotifications/model/SlackMessage.scala]: Done :+1:

File: app/uk/gov/hmrc/slacknotifications/services/AuthService.scala (hunk 3 above)
Comment (jennygrahamjones) [reply app/uk/gov/hmrc/slacknotifications/services/AuthService.scala]: Done :+1:

File: app/uk/gov/hmrc/slacknotifications/model/NotificationRequest.scala (hunk 1 above)
Comment (jennygrahamjones) [reply app/uk/gov/hmrc/slacknotifications/model/NotificationRequest.scala]: I've simplified this now

File: app/uk/gov/hmrc/slacknotifications/model/NotificationRequest.scala [hunk 4]
```diff
@@ -21,25 +21,29 @@ import play.api.libs.json.Reads._
...
-                                 iconEmoji: Option[String] = None,
-                                 attachments: Seq[Attachment] = Nil
-                               )
+  text: String,
+  username: String,
+  iconEmoji: Option[String]    = None,
+  attachments: Seq[Attachment] = Nil
+) {
+
+  def getFields: Array[String] = Array(this.text,this.username,this.iconEmoji.getOrElse(""))
```
Comment (konradwudkowski) [inline app/uk/gov/hmrc/slacknotifications/model/NotificationRequest.scala]: prefixing with `this` as in `this.text`, `this.username` is not needed

File: app/uk/gov/hmrc/slacknotifications/model/SlackMessage.scala [hunk 5]
```diff
@@ -22,30 +22,56 @@ import play.api.libs.json.{Json, OFormat}
...
   final case class Field(
-                          title: String,
-                          value: String,
-                          short: Boolean
-                        )
+    title: String,
+    value: String,
+    short: Boolean
+  ) {
+    def fields = Array(this.title, this.value)
```
Comment (konradwudkowski) [inline app/uk/gov/hmrc/slacknotifications/model/SlackMessage.scala]: prefixing with `this` as in `this.title`, `this.value` is not needed

File: app/uk/gov/hmrc/slacknotifications/model/NotificationRequest.scala [hunk 6]
```diff
@@ -21,25 +21,29 @@ import play.api.libs.json.Reads._
...
-                                 iconEmoji: Option[String] = None,
-                                 attachments: Seq[Attachment] = Nil
-                               )
+  text: String,
+  username: String,
+  iconEmoji: Option[String]    = None,
+  attachments: Seq[Attachment] = Nil
+) {
+
+  def getFields: Array[String] = Array(text, username, iconEmoji.getOrElse(""))
```
Comment (colin-lamed) [inline app/uk/gov/hmrc/slacknotifications/model/NotificationRequest.scala]: There's a lot of usage of Array which is mutable - prefer Vector or List.

Comment (jennygrahamjones) [reply app/uk/gov/hmrc/slacknotifications/model/NotificationRequest.scala]: I've made this change

**Pull request merged**